practice_project/weather_forecast.py

```diff
@@ -0,0 +1,86 @@
+import requests
+from abc import ABC, abstractmethod
+
+class WeatherDataProvider(ABC):
+    @abstractmethod
+    def fetch_weather_forecast(self, latitude, longitude):
+        """
+        Abstract method to fetch weather forecast for a given latitude and longitude.
+        """
+        pass
```

Comment:
especially for interface methods, it's highly recommended to add type-hints so that you know what data type the parameters and returned values have